custom_components/roborock/utils.py

```diff
@@ -7,7 +10,7 @@ def set_nested_dict(data: dict, key_string: str, value):
     here[keys[-1]] = value
 
 
-def get_nested_dict(data: dict, key_string: str, default=None):
+def get_nested_dict(data: dict, key_string: str, default=None) -> dict:
```

Comment:
This function confuses me. Would this not just return the value of the last key in key string in the data dict unless one of the values in None, in which case we return the default? Is that the intended result?

Reply:
It navigates the object tree. Like foo.bar.x will retries the value of x under {"foo":{"bar":{"x":1}}}